tests/test_errors.py

```diff
@@ -633,12 +634,73 @@ def test_loc_with_dots():
     )
 
 
-def subclass_validation_error():
+def test_subclass_validation_error():
     class MyValidationError(ValidationError):
-        def __init__(self, title: str, errors, error_mode='python'):
-            self.body = 123
-            super().__init__(title, errors, error_mode)
+        def __init__(self, message: str, errors, body: int = 123):
+            self.body = body
+            super().__init__(message, errors, 'python')
+
+        def derive(self, __exceptions: list[InitErrorDetails | ValidationError]) -> 'MyValidationError':
+            return MyValidationError(message=self.title, errors=__exceptions, body=self.body)
 
     e = MyValidationError('testing', [])
+    assert isinstance(e, MyValidationError)
     assert e.body == 123
     assert e.title == 'testing'
+
+    e2 = e.derive([{'type': 'greater_than', 'loc': ('a', 2), 'input': 4, 'ctx': {'gt': 5}}])
+    assert isinstance(e2, MyValidationError)
+    assert e2.body == 123
+    assert e2.title == 'testing'
+
+
+def test_raise_validation_tree():
+    class MyValidationError(ValidationError):
+        pass
+
+    with pytest.raises(ValidationError) as exc_info1:
+        raise ValidationError(
+            'Foobar', [{'type': 'greater_than', 'loc': ('a', 2), 'input': 4, 'ctx': {'gt': 5}}], 'python', (123,)
+        )
+
+    with pytest.raises(ValidationError) as exc_info2:
+        raise MyValidationError('Foobar', [{'type': 'greater_than', 'loc': ('a', 1), 'input': 2, 'ctx': {'gt': 5}}])
+
+    v = ValidationError(
+        'merged',
+        [exc_info1.value, exc_info2.value, {'type': 'less_than', 'loc': ('a', 2), 'input': 8, 'ctx': {'lt': 1}}],
+        'python',
+        ('outer',),
+    )
+
+    assert v.error_count() == 3
+    assert v.errors(include_url=False) == [
+        {'type': 'less_than', 'loc': (2, 'a'), 'msg': 'Input should be less than 1', 'input': 8, 'ctx': {'lt': 1}},
+        {
+            'type': 'greater_than',
+            'loc': (2, 'a'),
+            'msg': 'Input should be greater than 5',
+            'input': 4,
+            'ctx': {'gt': 5},
+        },
+        {
+            'type': 'greater_than',
+            'loc': (1, 'a'),
+            'msg': 'Input should be greater than 5',
+            'input': 2,
+            'ctx': {'gt': 5},
+        },
+    ]
+    # fmt: off
+    assert repr(v) == f"""\
+3 validation errors for merged
+2.a
+  Input should be less than 1 [type=less_than, input_value=8, input_type=int]
+    For further information visit https://errors.pydantic.dev/{__version__}/v/less_than
+2.a
+  Input should be greater than 5 [type=greater_than, input_value=4, input_type=int]
+    For further information visit https://errors.pydantic.dev/{__version__}/v/greater_than
+1.a
```

Comment:
Seems like the numbers are wrong here? I don't see any other tests for this. @samuelcolvin do you think it's a bug in this PR or in main?

Reply:
sorry, which number is wrong?

Reply:
I assume this should be 1a, 1b, 2?

Reply:
Oh, I see the order of the errors doesn't match the order in the list above - I think that must be related to this PR.

Reply:
in general errors are shown in the order of the vec they were added to. I think we should maintain that order.

Reply:
I’m pretty sure I am but will double check.